plugins/modules/site_workflow_manager.py

```diff
@@ -698,19 +893,34 @@ def site_requires_update(self):
             specified parameters, such as the site type and site details.
         """
 
-        type = self.have['current_site']['type']
-        updated_site = self.have['current_site']['site'][type]
-        requested_site = self.want['site_params']['site'][type]
+        if self.get_ccc_version_as_integer() <= self.version_2_3_5_3:
+            type = self.have.get('current_site', {}).get('type')
+            updated_site = self.have.get('current_site', {}).get('site', {}).get(type)
+            requested_site = self.want['site_params']['site'].get(type)
+
+        else:
+            type = self.have.get("current_site").get("type")
+            self.log("Type of site '{0}' for the site creation/updation in Cisco Catalyst Center.".format(type), "DEBUG")
+            current_site = self.have.get('current_site', {})
+            self.log("Current site details: {}".format(current_site), "INFO")
+            updated_site = current_site
+            requested_site = self.want.get('site_params', {}).get('site', {}).get(type)
+
         self.log("Current Site type: {0}".format(str(updated_site)), "INFO")
-        self.log("Requested Site type: {0}".format(str(requested_site)), "INFO")
+        self.log("Requested Site type: {0}".format(
+            str(requested_site)), "INFO")
 
```

Comment:
As we are returning in all the if statements, we can use if instead of elif. Shall we log a message before returning?
```
    if site_type == "building":
        needs_update = not self.is_building_updated(updated_site, requested_site)
        self.log("Building site requires update: {}".format(needs_update), "DEBUG")
        return needs_update

    if site_type == "floor":
        needs_update = not self.is_floor_updated(updated_site, requested_site)
        self.log("Floor site requires update: {}".format(needs_update), "DEBUG")
        return needs_update

    if site_type == "area":
        needs_update = not self.is_area_updated(updated_site, requested_site)
        self.log("Area site requires update: {}".format(needs_update), "DEBUG")
        return needs_update

    self.msg = "Unsupported site type '{0}' given in the playbook.".format(site_type)
    self.log(self.msg, "ERROR")
    self.status = "failed"
    return False 
```

Reply:
Yes, we can use
Added this part of code